StripePaymentSheet/StripePaymentSheet/Source/Internal/Link/Controllers/PayWithLinkWebController.swift

```diff
@@ -199,19 +202,25 @@ final class PayWithLinkWebController: NSObject, ASWebAuthenticationPresentationC
                 // Canceled for another reason - raise an error.
                 self.canceledWithError(error: error)
             }
-            self.payWithLinkDelegate?.payWithLinkWebControllerDidCancel(self)
             return
         }
         do {
             let result = try LinkPopupURLParser.result(with: returnURL)
             switch result.link_status {
             case .complete:
                 let paymentOption = PaymentOption.link(option: PaymentSheet.LinkConfirmOption.withPaymentMethod(paymentMethod: result.pm))
+
+                // Cache the PM details
+                let las = LinkAccountService()
+                las.setLastPMDetails(pm: result.pm)
+
                 STPAnalyticsClient.sharedClient.logLinkPopupSuccess(sessionType: self.context.intent.linkPopupWebviewOption)
                 self.payWithLinkDelegate?.payWithLinkWebControllerDidComplete(self, intent: self.context.intent, with: paymentOption)
             case .logout:
-                // We don't store any Link session information locally at the moment. Treat this as a cancel.
-                self.canceledWithoutError()
+                // Delete the account information
+                LinkAccountService.defaultCookieStore.clear()
+                STPAnalyticsClient.sharedClient.logLinkPopupLogout(sessionType: self.context.intent.linkPopupWebviewOption)
+                self.payWithLinkDelegate?.payWithLinkWebControllerDidCancel(self)
```

Comment:
Nit: Not sure if we care, but in this case of logging out we are calling:
```
clear()
loganalytic
didCancel()
```
but in canceledWithoutError():
```
analytic
clear()
didCancel()
```